monkey/infection_monkey/exploit/http_agent_binary_server_registrar.py

```diff
@@ -0,0 +1,34 @@
+from ipaddress import IPv4Address
+from typing import Optional, Sequence
+
+from patlib import PurePath
+
+from common import OperatingSystem
+
+from .http_agent_binary_server import (
+    AgentBinaryRequest,
+    HTTPAgentBinaryServer,
+    RequestID,
+    RequestType,
+)
+from .i_http_agent_binary_server_registrar import IHTTPAgentBinaryServerRegistrar
+
+
+class HTTPAgentBinaryServerRegistrar(IHTTPAgentBinaryServerRegistrar):
+    def __init__(self, server: HTTPAgentBinaryServer):
+        self._server = server
+
+    def register_request(
+        self,
+        operating_system: OperatingSystem,
+        request_type: RequestType,
+        requestor_ip: IPv4Address,
+        destination_path: Optional[PurePath] = None,
+        args: Sequence[str] = [],
+    ) -> AgentBinaryRequest:
+        return self._server.register(
+            operating_system, request_type, requestor_ip, destination_path, args
+        )
+
+    def deregister_request(self, request_id: RequestID):
+        self._server.unregister(request_id)
```

Comment:
What happens if the request ID isn't found? Where will the error be caught?

Reply:
@mssalvatore can correct me if I am wrong but that should happen when we use these methods